FIx the issue with the semantic release analyzer

Release jobs fail because the semantic-release process cannot find the module conventional-changelog-conventionalcommits, which is required by the @semantic-release/commit-analyzer plugin:

Error: Cannot find module 'conventional-changelog-conventionalcommits'

Solution

Add conventional-changelog-conventionalcommits as a dependency to your project.

If you use npm, run:

bash
npm install --save-dev conventional-changelog-conventionalcommits
If you use yarn, run:

bash
yarn add --dev conventional-changelog-conventionalcommits
After adding the dependency, commit your updated package.json and package-lock.json (or yarn.lock) files, and push to your repository. This should resolve the MODULE_NOT_FOUND error and allow the workflow to proceed.

--- a/package.json
+++ b/package.json
@@ -6,8 +6,11 @@
     "devDependencies": {
         "semantic-release": "^22.0.0",
         "@semantic-release/changelog": "^6.0.3",
+        "@semantic-release/commit-analyzer": "^11.1.0",
         "@semantic-release/git": "^10.0.1",
-        "@semantic-release/github": "^9.2.6"
+        "@semantic-release/github": "^9.2.6",
+        "@semantic-release/release-notes-generator": "^12.1.0",
+        "conventional-changelog-conventionalcommits": "^7.0.2"
     },
     "scripts": {
         "semantic-release": "semantic-release"

